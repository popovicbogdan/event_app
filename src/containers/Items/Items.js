import React, { useEffect, useState, useCallback } from 'react';
import { useDispatch, useSelector, connect } from 'react-redux';
import { orderBy } from 'lodash';

import { Container, Hr, ButtonStyled, IconsContainer } from './Items.styled';
import Section from '../Section/Section';
import Title from '../../components/Title/Title';
import Icon from '../../components/Icon/Icon';
import svgContainer from '../../components/Icon/svgContainer';
import { Facebook, Twitter, Instagram } from '../../components/Icon/icons';
import { createSelector } from 'reselect';
import { filteredItemsSelector } from '../../store/selectors/selectors';

const Items = ({ items }) => {
  const [viewOne, setViewOne] = useState('grid');
  const [viewTwo, setViewTwo] = useState('list');

  const changeViewType = (view, setView) => {
    switch (view) {
      case 'grid':
        setView('list');
        break;
      default:
        setView('grid');
    }
  };

  // const sortAscending = useCallback(() => {
  //   const sortedItems = orderBy(items, ["width", "height"]);
  //   setItms(sortedItems);
  // }, [items]);

  // const sortDescending = useCallback(() => {
  //   const sortedItems = orderBy(items, ["width", "height"]).reverse();
  //   setItms(sortedItems);
  // }, [items]);

  // const filterItems = (items, searchTerm) => {
  //   const filteredItems = items.filter(item =>
  //     Object.values(item).some(val =>
  //       val
  //         .toString()
  //         .toLowerCase()
  //         .includes(searchTerm.toLowerCase())
  //     )
  //   );
  //   return filteredItems;
  // };
  // const sectionOneItems = filterItems(items.slice(0, 6), search);
  const sectionOneItems = items.slice(0, 6);

  const sectionTwoItems = items.slice(6, 9);
  // const sectionTwoItems = filterItems(items.slice(6, 9), search);

  console.log('RENDERED');

  return (
    <Container>
      <Title>Hot tickets</Title>
      <Hr />
      <Section items={sectionOneItems} viewType={viewOne} />
      <ButtonStyled
        className="btn btn-small right"
        onClick={(e) => changeViewType(viewOne, setViewOne)}
      >
        <Icon>{viewOne === 'grid' ? 'list' : 'grid_on'}</Icon>
      </ButtonStyled>
      <Title>Upcoming events</Title>
      <Hr />
      <Section items={sectionTwoItems} viewType={viewTwo} />
      <ButtonStyled
        className="btn btn-small right"
        onClick={(e) => changeViewType(viewTwo, setViewTwo)}
      >
        <Icon>{viewTwo === 'grid' ? 'list' : 'grid_on'}</Icon>
      </ButtonStyled>
      {/* <ButtonStyled className="btn btn-small right" onClick={sortAscending}>
        asc
      </ButtonStyled>

      <ButtonStyled className="btn btn-small right" onClick={sortDescending}>
        desc
      </ButtonStyled> */}

      <IconsContainer>
        <Facebook />
        <Instagram />
        <Twitter />
      </IconsContainer>
    </Container>
  );
};
const mapStateToProps = (state) => ({
  items: filteredItemsSelector(state)
});

export default connect(mapStateToProps)(Items);
