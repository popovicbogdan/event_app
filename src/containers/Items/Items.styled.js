import styled from "styled-components";
import Button from "../../components/Button/Button";

export const Container = styled.div.attrs({ className: "container" })`
  background: white;
  @media (max-width: 600px) {
    width: 70%;
  }
`;

export const Hr = styled.hr`
  margin-left: 10px;
  border-width: 1.5px;
  border-color: #e838ff;
  width: 5%;
`;
export const ButtonStyled = styled(Button)`
  margin-right: 10px;
  margin-bottom: 15px;
  height: 36px;
`;
