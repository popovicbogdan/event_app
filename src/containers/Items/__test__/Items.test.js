import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createStore } from "redux";
import renderer from "react-test-renderer";
import Items from "../Items";
import React from "react";
import { Provider } from "react-redux";

configure({ adapter: new Adapter() });
describe("<Items />", () => {
  const store = createStore(() => ({
    items: {
      items: [
        { id: 1, author: "Brandon Cooper", height: 1080, width: 1900 },
        { id: 2, author: "Alison Burgers", height: 1180, width: 1912 },
        { id: 3, author: "Kent Cooper", height: 1230, width: 1915 }
      ],
      searchInfo: ""
    }
  }));
  const component = renderer
    .create(
      <Provider store={store}>
        <Items />
      </Provider>
    )
    .toJSON();
  it("should match snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  const mockStore = [
    { id: 1, author: "Brandon Cooper", height: 1080, width: 1900 },
    { id: 2, author: "Alison Burgers", height: 1180, width: 1912 },
    { id: 3, author: "Kent Cooper", height: 1230, width: 1915 }
  ];

  it("filters the items", () => {
    const filtFunc = jest.fn((items, searchTerm) => {
      const filteredItems = items.filter(item =>
        Object.values(item).some(val =>
          val
            .toString()
            .toLowerCase()
            .includes(searchTerm.toLowerCase())
        )
      );
      return filteredItems;
    });
    expect(filtFunc(mockStore, "1080")).toEqual([
      {
        id: 1,
        author: "Brandon Cooper",
        height: 1080,
        width: 1900
      }
    ]);
    expect(filtFunc).toHaveBeenCalledTimes(1);
  });
});
