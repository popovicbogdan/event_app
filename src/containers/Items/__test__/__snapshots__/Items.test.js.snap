// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Items /> should match snapshot 1`] = `
<div
  className="sc-AykKE dGIjfh container"
>
  <h5
    className="sc-fzXfMx oSHWh"
  >
    <nobr>
      Hot tickets
    </nobr>
  </h5>
  <hr
    className="sc-AykKF bFqZwE"
  />
  <div
    className="sc-fzXfMw jtMUjW row"
  >
    <div
      className="col s12 m6 l4"
    >
      <a>
        <div
          className="sc-AykKH ihxFCz card hoverable"
        >
          <div
            className="sc-AykKI grNqlY card-image"
          >
            <img
              alt=""
              className="sc-AykKJ iuCgpc"
            />
            <p
              className="author"
            >
              Brandon Cooper
            </p>
          </div>
          <div
            className="sc-AykKK gkWpsO card-content"
          >
            <p>
              <i
                className="sc-AykKD fZGMjR material-icons"
                data-testid="icon"
              >
                event
              </i>
              <span
                className="height"
              >
                Height: 
                1080
              </span>
            </p>
            <p>
              <i
                className="sc-AykKD fZGMjR material-icons"
                data-testid="icon"
              >
                deck
              </i>
              <span
                className="width"
              >
                Width: 
                1900
              </span>
            </p>
          </div>
        </div>
      </a>
    </div>
    <div
      className="col s12 m6 l4"
    >
      <a>
        <div
          className="sc-AykKH ihxFCz card hoverable"
        >
          <div
            className="sc-AykKI grNqlY card-image"
          >
            <img
              alt=""
              className="sc-AykKJ iuCgpc"
            />
            <p
              className="author"
            >
              Alison Burgers
            </p>
          </div>
          <div
            className="sc-AykKK gkWpsO card-content"
          >
            <p>
              <i
                className="sc-AykKD fZGMjR material-icons"
                data-testid="icon"
              >
                event
              </i>
              <span
                className="height"
              >
                Height: 
                1180
              </span>
            </p>
            <p>
              <i
                className="sc-AykKD fZGMjR material-icons"
                data-testid="icon"
              >
                deck
              </i>
              <span
                className="width"
              >
                Width: 
                1912
              </span>
            </p>
          </div>
        </div>
      </a>
    </div>
    <div
      className="col s12 m6 l4"
    >
      <a>
        <div
          className="sc-AykKH ihxFCz card hoverable"
        >
          <div
            className="sc-AykKI grNqlY card-image"
          >
            <img
              alt=""
              className="sc-AykKJ iuCgpc"
            />
            <p
              className="author"
            >
              Kent Cooper
            </p>
          </div>
          <div
            className="sc-AykKK gkWpsO card-content"
          >
            <p>
              <i
                className="sc-AykKD fZGMjR material-icons"
                data-testid="icon"
              >
                event
              </i>
              <span
                className="height"
              >
                Height: 
                1230
              </span>
            </p>
            <p>
              <i
                className="sc-AykKD fZGMjR material-icons"
                data-testid="icon"
              >
                deck
              </i>
              <span
                className="width"
              >
                Width: 
                1915
              </span>
            </p>
          </div>
        </div>
      </a>
    </div>
  </div>
  <button
    className="sc-AykKC hBViMd sc-AykKG hDsgAj btn btn-small right"
    data-testid="button"
    onClick={[Function]}
    type="button"
  >
    <i
      className="sc-AykKD fZGMjR material-icons"
      data-testid="icon"
    >
      list
    </i>
  </button>
  <h5
    className="sc-fzXfMx oSHWh"
  >
    <nobr>
      Upcoming events
    </nobr>
  </h5>
  <hr
    className="sc-AykKF bFqZwE"
  />
  <ul />
  <button
    className="sc-AykKC hBViMd sc-AykKG hDsgAj btn btn-small right"
    data-testid="button"
    onClick={[Function]}
    type="button"
  >
    <i
      className="sc-AykKD fZGMjR material-icons"
      data-testid="icon"
    >
      grid_on
    </i>
  </button>
  <button
    className="sc-AykKC hBViMd sc-AykKG hDsgAj btn btn-small right"
    data-testid="button"
    onClick={[Function]}
    type="button"
  >
    asc
  </button>
  <button
    className="sc-AykKC hBViMd sc-AykKG hDsgAj btn btn-small right"
    data-testid="button"
    onClick={[Function]}
    type="button"
  >
    desc
  </button>
</div>
`;
